### PR TITLE
Bible text size buttons

### PR DESCRIPTION
**Pull request by zhuiks**
Closing #42, already merged with #195. Please close this only after #195 

That's how it looks:
Screenshot from 2021-03-19 10-17-22

### CODE REVIEW

**Comment by tobias-klein** [comment]
Can't look at this immediately ... but within the next couple of days!
Thanks for your efforts, I appreciate it!

**Comment by tobias-klein** [comment]
I have just had a first look from a user perspective. It works well - including persistence!

One question that came to my mind though ... wouldn't it make sense to also increase the text-size of the following items simultaneously:
* Verse Reference text (may require also adjusting the overall width of that column, especially for passages like Psalm 119)
* Verse tags
* Verse notes
* Book notes

What do you think?

I'll have a separate look regarding the coding of this.

**Comment by zhuiks** [comment]
> One question that came to my mind though ... wouldn't it make sense to also increase the text-size of the following items simultaneously:
> 
>     * Verse Reference text (may require also adjusting the overall width of that column, especially for passages like Psalm 119)
>     * Verse tags
>     * Verse notes
>     * Book notes
> 
> What do you think?

I think it depends on use cases. I personally prefer to have Bible text bigger as it's easier to concentrate on that. Maybe it can be another display option: "Adjust tags and notes text size along with Bible text".
In any case it's easy to add as current implementation just simply updates css rule.

> Another thing that comes to my mind ... How about adding shortcuts for increasing and decreasing the font-size. Typically it would be CTRL/CMD +/- (Win/Linux vs. Mac). Search the code for `Mousetrap.bind` and you'll find some examples!

Great idea, I'll look into that.

**Comment by tobias-klein** [comment]
> > One question that came to my mind though ... wouldn't it make sense to also increase the text-size of the following items simultaneously:
> > 
> > * Verse Reference text (may require also adjusting the overall width of that column, especially for passages like Psalm 119)
> > * Verse tags
> > * Verse notes
> > * Book notes
> > 
> > What do you think?
> 
> I think it depends on use cases. I personally prefer to have Bible text bigger as it's easier to concentrate on that. Maybe it can be another display option: "Adjust tags and notes text size along with Bible text".
> In any case it's easy to add as current implementation just simply updates css rule.

The Bible text is bigger than the other texts (notes/tags) in general.
However, I guess if somebody uses the app on a really large screen there may be a motivation to adjust all text content in the bible text container. And yes, we could make that configurable in the options menu, good idea!

File: app/frontend/components/text_size_settings.js [hunk 1]
```diff
@@ -1,31 +1,79 @@
...
-    this._settingsObject = window.ipcSettings;
+    this.defaultTextSize = 10; // in em*10 so not to deal with float
+    this.minSize = 7;
+    this.maxSize = 20;
+    this._textSizeValue = this.defaultTextSize;
+    this._settingsKey = 'verse-text-size';
     this.openMenuButton = '.text-size-settings';
     this.menuContainer = '.text-size-menu';
     this.menuIsOpened = false;
+    this.stylesheet = null;  // CSSStyleSheet
```
Comment (tobias-klein) [inline app/frontend/components/text_size_settings.js:11]: I suggest to add a space in the comment for readability ...
`CSS Stylesheet`

File: app/frontend/components/text_size_settings.js [hunk 2]
```diff
@@ -1,31 +1,79 @@
...
 
-    var top_offset = $menuButton.top + $menuButton.height() + 1;
-    var left_offset = $menuButton.left + $menuButton.width()/2 - $menuContainer.width()/2;    
+    var top_offset = buttonOffset.top + $menuButton.height() + 2;
+    var left_offset = buttonOffset.left - 2;
+    if(left_offset+$menuContainer.width() > $(window).width()) {
+      left_offset = buttonOffset.left + $menuButton.width() - $menuContainer.width();
+    }
+
+    console.log(`[${buttonOffset}] ->[${top_offset}, ${left_offset}]`);
```
Comment (tobias-klein) [inline app/frontend/components/text_size_settings.js:76]: I suggest to out comment the console.log before merging.

File: app/frontend/components/text_size_settings.js [hunk 3]
```diff
@@ -1,31 +1,79 @@
...
+    var $menuButton = currentVerseListMenu.find(this.openMenuButton);
     $menuButton.addClass('ui-state-active');
+    var buttonOffset = $menuButton.offset();
     var $menuContainer = currentVerseListMenu.find(this.menuContainer);
 
-    var top_offset = $menuButton.top + $menuButton.height() + 1;
-    var left_offset = $menuButton.left + $menuButton.width()/2 - $menuContainer.width()/2;    
+    var top_offset = buttonOffset.top + $menuButton.height() + 2;
+    var left_offset = buttonOffset.left - 2;
+    if(left_offset+$menuContainer.width() > $(window).width()) {
```
Comment (tobias-klein) [inline app/frontend/components/text_size_settings.js:72]: Code style: Insert a space - `if (`

File: app/frontend/components/text_size_settings.js [hunk 4]
```diff
@@ -1,31 +1,79 @@
...
-    var $menuButton = currentVerseListMenu.find('.display-options-button');
+    var $menuButton = currentVerseListMenu.find(this.openMenuButton);
     $menuButton.addClass('ui-state-active');
+    var buttonOffset = $menuButton.offset();
     var $menuContainer = currentVerseListMenu.find(this.menuContainer);
 
-    var top_offset = $menuButton.top + $menuButton.height() + 1;
-    var left_offset = $menuButton.left + $menuButton.width()/2 - $menuContainer.width()/2;    
+    var top_offset = buttonOffset.top + $menuButton.height() + 2;
+    var left_offset = buttonOffset.left - 2;
```
Comment (tobias-klein) [inline app/frontend/components/text_size_settings.js:71]: Code style: Let's try to be consistent using camel case for all variables. There is some legacy code where this is not the case, but let's try to use this style for all new code.

File: app/frontend/components/text_size_settings.js [hunk 5]
```diff
@@ -1,31 +1,79 @@
 class TextSizeSettings {
```
Comment (tobias-klein) [inline app/frontend/components/text_size_settings.js:1]: Please add a small API docs summary above the class definition

**Review by tobias-klein — COMMENTED** [review]
Thanks for this!

I only have a few comments, mostly style-related. Your code is easy to read and clean! 👍

File: html/verse_list_tabs.html [hunk 6]
```diff
@@ -1,79 +1,119 @@
...
+      <div class="text-size-menu verse-selection-menu" style="display: none;">
+        <div class="verse-list-menu-line">
+          <div class="fg-button text-size-decrease ui-state-default ui-corner-all" i18n="[title]menu.text-size-decrease">
+            <i class="fas fa-font"></i><i class="fas fa-minus fa-xs"></i>
+          </div>
+          <div class="fg-button text-size-reset ui-state-default ui-corner-all" i18n="[title]menu.text-size-reset">
+            <i class="fas fa-font"></i>
+          </div>
+          <div class="fg-button text-size-increase ui-state-default ui-corner-all" i18n="[title]menu.text-size-increase">
+            <i class="fas fa-font"></i><i class="fas fa-plus fa-xs"></i>
```
Comment (tobias-klein) [inline html/verse_list_tabs.html:47]: Would you mind adding the new i18n texts to all the locales files?

**Comment by zhuiks** [comment]
I added all those changes we were talking about including keyboard shortcuts and display option.
@tobias-klein what should adjust text size buttons dialog do when user switches to another tab or clicks somewhere else?

**Comment by tobias-klein** [comment]
Thank you! Excellent!
I think the menu should simply be hidden when clicking somewhere else, also when switching to another tab. I believe there is already functionality like this implemented for other menus and there is a central function that hides all menus.

**Comment by zhuiks** [comment]
> I think the menu should simply be hidden when clicking somewhere else, also when switching to another tab. I believe there is already functionality like this implemented for other menus and there is a central function that hides all menus.

This should do it

**Comment by tobias-klein** [comment]
Excellent!

**Comment by tobias-klein** [comment]
It works very well! Excellent job.

I have one more suggestion. We can disable/enable the buttons (add/remove class `ui-state-disabled`) for the following cases:

* Initial state: The 'reset to default' button can be disabled, because it is the default already. Once changing the font-size the button can be enabled, once resetting it disabled again.
* Once reaching the maximum font-size the increase button can be disabled. Once going down a notch it can be enabled again.
* Once reaching the minimum the decrease button can be disabled. Once going up a notch it can be enabled again.

**Comment by zhuiks** [comment]
> I have one more suggestion. We can disable/enable the buttons (add/remove class `ui-state-disabled`)

Great idea! Here you go: 033f0ed

**Review by tobias-klein — APPROVED** [review]

**Pull request merged**